SWATCH-2113 Add internal API to update Billable Usage remittance value

kartikshahc opened the pull request:
<!-- Replace XXXX with the issue number. Issue will be auto-linked -->
Jira issue: SWATCH-2113

## Description
<!-- Provide a description of this PR.  Try to provide answers to "what", "how",
and "why" -->

## Testing
<!--
When possible, please use commands a developer can directly paste or modify
-->

### Setup
<!-- Add any steps required to set up the test case -->
1.
1.

### Steps
<!-- Enter each step of the test below -->
1.
1.

### Verification
<!-- Enter the steps needed to verify the test passed -->
1.
1.

mstead reviewed (changes requested):
This works as expected and will provide the functionality we need.

I would expect to see this code hang around until we get the DLQ/retry functionality in place for our billable usage processors. Even though this may be throw away, lets rename the API call to reflect that we are resetting the value of remittance, instead of simply a PUT to billable_usage.

We should get a couple tests around some of this logic since it will likely be around for a short while.

kartikshahc commented:
> This works as expected and will provide the functionality we need.
> 
> I would expect to see this code hang around until we get the DLQ/retry functionality in place for our billable usage processors. Even though this may be throw away, lets rename the API call to reflect that we are resetting the value of remittance, instead of simply a PUT to billable_usage.
> 
> We should get a couple tests around some of this logic since it will likely be around for a short while.

@mstead I have added a failing test and a non failing test in a method. Also I have renamed the API endpoint to `reset_billable_usage_remittance`. Please let me know it looks good to you. Thanks!

mstead commented on src/main/spec/internal-billing-api-spec.yaml:
```diff
@@ -83,6 +83,52 @@ paths:
         - internalBilling
+  /internal/rpc/remittance/reset_billable_usage_remittance:
+    put:
+      summary: Update billable usage remittance records.
```
Please add some details of what it means to 'reset' billable usage.

kartikshahc replied:
Done

mstead reviewed (changes requested):
One more tiny request and it'll be good to go.
